### PR TITLE
docs: Add advanced configs guide

### PR DESCRIPTION
**Pull request by Jeffail**

### CODE REVIEW

**Comment by binarylogic** [comment]
Let me know when you want me to review

File: website/docs/setup/guides/advanced-configs.md [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+sidebar_label: Advanced Configs
+description: Strategies for building and maintaining complex Vector configs
+---
+
+Writing large configuration files is not yet an official olympic event. However,
+it's still a good idea to get yourself ahead of the competition. In this guide
+we're going to cover some tips and tricks that will help you write clear, bug
+free Vector configs that are easy to maintain.
+
+## Building Pipeline Structure
```
Comment (Hoverbear) [inline website/docs/setup/guides/advanced-configs.md]: Maybe:

```suggestion
## Building pipelines
```

File: website/docs/setup/guides/advanced-configs.md [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+In cases where we simply want to chain a few components together we can avoid
+writing most of the boilerplate with the `generate` subcommand. This allows us
+to express a list of components, where it generates a config with all of those
+components connected in a linear chain.
+
+For example, if we wished to create a chain of three transforms; `json_parser`,
+`add_fields`, and `remove_fields`, we can run:
+
+```bash
+vector generate /json_parser,add_fields,remove_fields > example.toml
```
Comment (Hoverbear) [inline website/docs/setup/guides/advanced-configs.md]: One thing I've noticed, it's not useful to do something like 

```bash
$ vector generate file >> example.toml
$ vector generate /json_parser,add_fields,remove_fields >> example.toml
$ vector generate //console >> example.toml
```

Due to the globals being dropped in. Maybe it's worth having a `--fragment` flag?

File: website/docs/setup/guides/advanced-configs.md [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+  inputs = [ "transform1" ]
+  type = "remove_fields"
+  # etc ...
+```
+
+The names of the generated components are sequential (`transform0`,
+`transform1`, and so on). It's therefore worth doing a search and replace with
+your editor to give them better names, e.g. `s/transform2/scrub_emails/g`.
+
+### Swimlanes
```
Comment (binarylogic) [inline website/docs/setup/guides/advanced-configs.md]: This seems a little out of place to me. This is more about building topologies and controlling flow, and less about config development and maintenance. Do you agree?

**Comment by binarylogic** [comment]
Noting: I think it's worth mentioning the ability to reload configs on the fly with Vector as well as the new `--watch-config` flag. Watching should only be used in environments where users can't issue a `SIGHUP` to Vector easily (like k8s). Otherwise, I find explicitly sending a `SIGHUP` to be much safer.

File: website/docs/setup/guides/advanced-configs.md (hunk 2 above)
Comment (Jeffail) [reply website/docs/setup/guides/advanced-configs.md]: Yeah agreed, opened https://github.com/timberio/vector/issues/1851

File: website/docs/setup/guides/advanced-configs.md (hunk 3 above)
Comment (Jeffail) [reply website/docs/setup/guides/advanced-configs.md]: Yeah I'm not attached to it.

File: website/docs/setup/guides/advanced-configs.md [hunk 4]
```diff
@@ -180,4 +180,5 @@ $ vector -c ./configs/foo.toml ./configs/bar.toml
 If you have a large chain of components it's a good idea to break them out into
 individual files, each with its own unit tests.
 
+
```
Comment (Hoverbear) [inline website/docs/setup/guides/advanced-configs.md]: This newline:

**Comment by Jeffail** [comment]
I've moved the sections around a little bit as I really want to emphasize test driven configuration.

**Comment by binarylogic** [comment]
Nice work on this 👍

**Pull request merged**